src/plone/app/multilingual/google_translate.py

```diff
@@ -0,0 +1,71 @@
+from plone.app.multilingual.interfaces import IMultiLanguageExtraOptionsSchema
+from plone.registry.interfaces import IRegistry
+from zope.component import getUtility
+
+import json
+import urllib
+
+
+class GoogleCloudTranslationAPIFactory:
+    """implement the external translation using Google Cloud Translation API"""
+
+    order = 999
```

Comment:
An idea might be to define a schema for translation service controlpanel with “enabled”, “order”, “available_source_languages”, and “available_target_languages” fields.

Reply:
Yes, I tried to keep backwards compatibility in thilese changes and that's why I didn't change the controlpanel at all.

But what you say makes total sense. The point is: should we extract the google translate support from p.a.multilingual and put it in an addon?